tests/ops/qutrit/test_qutrit_parametric_ops.py

```diff
@@ -0,0 +1,384 @@
+ALL_OPERATIONS = NON_PARAMETRIZED_OPERATIONS + PARAMETRIZED_OPERATIONS
+
+dot_broadcasted = lambda a, b: np.einsum("...ij,...jk->...ik", a, b)
+multi_dot_broadcasted = lambda matrices: reduce(dot_broadcasted, matrices)
+
+
+class TestOperations:
+    @pytest.mark.parametrize("op", ALL_OPERATIONS + BROADCASTED_OPERATIONS)
+    def test_parametrized_op_copy(self, op, tol):
+        """Tests that copied parametrized ops function as expected"""
+        copied_op = copy.copy(op)
+        assert np.allclose(op.matrix(), copied_op.matrix(), atol=tol)
+
+        op.inv()
+        copied_op2 = copy.copy(op)
+        assert np.allclose(op.matrix(), copied_op2.matrix(), atol=tol)
+        op.inv()
+
+    @pytest.mark.parametrize("op", PARAMETRIZED_OPERATIONS)
+    def test_adjoint_unitaries(self, op, tol):
+        """Test that matrices of adjoint operations behave correctly"""
+        op_d = op.adjoint()
+        res1 = np.dot(op.matrix(), op_d.matrix())
+        res2 = np.dot(op_d.matrix(), op.matrix())
+        assert np.allclose(res1, np.eye(3 ** len(op.wires)), atol=tol)
+        assert np.allclose(res2, np.eye(3 ** len(op.wires)), atol=tol)
+        assert op.wires == op_d.wires
+
+    @pytest.mark.parametrize("op", BROADCASTED_OPERATIONS)
+    def test_adjoint_unitaries_broadcasted(self, op, tol):
+        """Test that matrices of adjoint operations with broadcasting behave correctly"""
+        op_d = op.adjoint()
+        res1 = dot_broadcasted(op.matrix(), op_d.matrix())
+        res2 = dot_broadcasted(op_d.matrix(), op.matrix())
+        I = [np.eye(3 ** len(op.wires))] * op.batch_size
+        assert np.allclose(res1, I, atol=tol)
+        assert np.allclose(res2, I, atol=tol)
+        assert op.wires == op_d.wires
+
+
+class TestParameterFrequencies:
+    @pytest.mark.parametrize("op", PARAMETRIZED_OPERATIONS)
+    def test_parameter_frequencies_match_generator(self, op, tol):
+        """Check that parameter frequencies of parametrized operations are defined correctly."""
+        if not qml.operation.has_gen(op):
+            pytest.skip(f"Operation {op.name} does not have a generator defined to test against.")
+
+        gen = op.generator()
+
+        try:
+            mat = gen.matrix()
+        except (AttributeError, qml.operation.MatrixUndefinedError):
+
+            if isinstance(gen, qml.Hamiltonian):
+                mat = qml.utils.sparse_hamiltonian(gen, level=3).toarray()
+            elif isinstance(gen, qml.SparseHamiltonian):
+                mat = gen.sparse_matrix().toarray()
+            else:
+                pytest.skip(f"Operation {op.name}'s generator does not define a matrix.")
+
+        gen_eigvals = np.round(np.linalg.eigvalsh(mat), 8)
+        freqs_from_gen = qml.gradients.eigvals_to_frequencies(tuple(gen_eigvals))
+
+        freqs = op.parameter_frequencies
+        assert np.allclose(freqs, freqs_from_gen, atol=tol)
+
+
+# TODO: Add tests for decompositions
+
+
+class TestMatrix:
+    def test_trx(self, tol):
```

Comment:
Could we make use of `pytest.mark.parameterize` here? Instead of having multiple asserts, just a single assert and a list of inputs instead.

Reply:
Good catch. I got too caught up making sure I follow the same structure as the qubit parametric op test file. I'll fix this now.